File: app/src/main/java/com/murile/jokenpo/MainActivity.java
package com.murile.jokenpo;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    public void selectRock(View view) {
        checkWinner("rock");
    }

    public void selectPaper(View view) {
        checkWinner("paper");

    }

    public void selectScissors(View view) {
        checkWinner("scissors");
    }

    private String generateAppChoice() {
        int randomN = (int) (Math.random() * 3);
        String[] options = {"rock", "paper", "scissors"};
        String appChoice = options[randomN];
        TextView textChoice = findViewById(R.id.textChoice);
        switch (appChoice) {
            case "rock":
                textChoice.setText(R.string.rock);
                break;
            case "paper":
                textChoice.setText(R.string.paper);
                break;
            case "scissors":
                textChoice.setText(R.string.scissors);
                break;
        }
        return appChoice;
    }

    private String getLocalizedChoice(String choice) {
        switch (choice) {
            case "rock":
                return getString(R.string.rock_text);
            case "paper":
                return getString(R.string.paper_text);
            case "scissors":
                return getString(R.string.scissors_text);
            default:
                return "choice";
        }
    }

    private void checkWinner(String userChoice) {
        String appChoice = generateAppChoice();
        TextView textResult = findViewById(R.id.textResult);
        TextView textOverview = findViewById(R.id.textOverview);

        String localizedUserChoice = getLocalizedChoice(userChoice);
        String localizedAppChoice = getLocalizedChoice(appChoice);

        textOverview.setText(String.format(getString(R.string.game_result), localizedUserChoice, localizedAppChoice));

        if ((appChoice.equals("rock") && userChoice.equals("scissors"))
                || (appChoice.equals("paper") && userChoice.equals("rock"))
                || (appChoice.equals("scissors") && userChoice.equals("paper"))) {
            textResult.setText(R.string.you_lost);
        } else if ((appChoice.equals("rock") && userChoice.equals("paper"))
                || (appChoice.equals("paper") && userChoice.equals("scissors"))
                || (appChoice.equals("scissors") && userChoice.equals("rock"))) {
            textResult.setText(R.string.you_won);
        } else {
            textResult.setText(R.string.draw);
        }

    }
}

File: app/src/main/java/com/murile/jokenpo/MainActivity.kt
package com.murile.jokenpo

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
    fun selectRock(view: View) {
        checkWinner("rock")
    }

    fun selectPaper(view: View) {
        checkWinner("paper")
    }

    fun selectScissors(view: View) {
        checkWinner("scissors")
    }

    private fun generateAppChoice(): String {
        val randomNumber = (Math.random() * 3).toInt()
        val options = arrayOf("rock", "paper", "scissors")
        val appChoice = options[randomNumber]
        val textChoice = findViewById<TextView>(R.id.textChoice)
        val textApp = findViewById<TextView>(R.id.textApp)

        when (appChoice) {
            "rock" -> {
                textApp.setText(R.string.app_choice_rock)
                textChoice.setText(R.string.rock)
            }
            "paper" -> {
                textApp.setText(R.string.app_choice_paper)
                textChoice.setText(R.string.paper)
            }
            "scissors" -> {
                textApp.setText(R.string.app_choice_scissors)
                textChoice.setText(R.string.scissors)
            }
        }
        return appChoice
    }

    private fun getLocalizedChoice(choice: String): String {
        return when (choice) {
            "rock" -> getString(R.string.rock_text)
            "paper" -> getString(R.string.paper_text)
            "scissors" -> getString(R.string.scissors_text)
            else -> choice
        }
    }

    private fun checkWinner(userChoice: String) {
        val appChoice = generateAppChoice()
        val textResult = findViewById<TextView>(R.id.textResult)
        val textOverview = findViewById<TextView>(R.id.textOverview)

        val localizedUserChoice = getLocalizedChoice(userChoice)
        val localizedAppChoice = getLocalizedChoice(appChoice)

        textOverview.text = String.format(getString(R.string.game_result), localizedUserChoice, localizedAppChoice)

        when {
            (appChoice == "rock" && userChoice == "scissors") ||
                    (appChoice == "paper" && userChoice == "rock") ||
                    (appChoice == "scissors" && userChoice == "paper") -> {
                textResult.setText(R.string.you_lost)
            }
            (appChoice == "rock" && userChoice == "paper") ||
                    (appChoice == "paper" && userChoice == "scissors") ||
                    (appChoice == "scissors" && userChoice == "rock") -> {
                textResult.setText(R.string.you_won)
            }
            else -> {
                textResult.setText(R.string.draw)
            }
        }
    }
}